salt/states/zpool.py

```diff
@@ -30,6 +30,23 @@
               /dev/disk2
               /dev/disk3
 
+    partitionpool:
+      zpool.present:
+        - config:
+            import: false
+            force: true
+        - properties:
+            comment: disk partition salty storage pool
+            ashift: '12'
+            feature@lz4_compress: enabled
+        - filesystem_properties:
+            compression: lz4
+            atime: on
+            relatime: on
+        - layout:
+            disk-0:
```

Comment:
Yes, that works for me as well.  So the issue as you've pointed out in your following comment, is with the need for the dummy `disk-0`.  Thanks for all of your time testing this out.